src/App/AdminProfile/AdminProfile.java

```diff
@@ -0,0 +1,170 @@
+import javafx.fxml.Initializable;
+import javafx.fxml.FXML;
+import javafx.scene.Parent;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.stage.Stage;
+
+import java.net.URL;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.*;
+import java.util.ResourceBundle;
+
+public class AdminProfile implements Initializable {
+    private int adminId;
+    private String UserRole;
+
+    public String getUserRole(){ return UserRole; }
+
+    public void setUserRole(String userRole){ this.UserRole=userRole; }
+
+    public int getAdminId(){ return adminId; }
+
+    public void setAdminId(int adminid){ this.adminId=adminid; }
+
+    public JFXButton homeBackBtn;
+    public JFXButton editProfileBtn;
+    public JFXButton saveBtn;
+    public JFXTextField adminIdText;
+    public JFXTextField adminNameText;
+    public JFXTextField adminUsernameText;
+    public JFXTextField adminPassText;
+    public JFXTextField adminDesText;
+    public Label confirmationMsg;
+
+    public void setAdminInfo() throws SQLException {
+        Connect connect = new Connect();
+        Connection connection = connect.getConnection();
+        Statement statement = connection.createStatement();
+
+        String sql = "SELECT * FROM ADMIN WHERE id="+adminId;
+        String sql2 = "SELECT * FROM ADMIN_AUTH WHERE id="+adminId;
+
+        ResultSet rs = statement.executeQuery(sql);
+
+        while(rs.next()){
+            int id = rs.getInt("id");
+            String name = rs.getString("name");
+            String desig = rs.getString("designation");
+
+            adminNameText.setText(name);
+            adminIdText.setText(String.valueOf(id));
+            adminDesText.setText(desig);
+        }
+
+        ResultSet rs2 = statement.executeQuery(sql2);
+
+        while(rs2.next()){
+            String username = rs2.getString("username");
+            String pass = rs2.getString("password");
+
+            adminUsernameText.setText(username);
+            adminPassText.setText(pass);
+        }
+    }
+
+    @Override
+    public void initialize(URL location, ResourceBundle resources) {
+        saveBtn.setVisible(false);
+        saveBtn.setDisable(true);
+
+        adminNameText.setEditable(false);
+        adminPassText.setEditable(false);
+        adminDesText.setEditable(false);
+    }
+
+    public void homeBackBtnAction(ActionEvent event) {
+        if(event.getSource() == homeBackBtn){
+            FXMLLoader Loader = new FXMLLoader();
+
+            Loader.setLocation(getClass().getResource("../IntroPageAdmin/IntroPageAdmin.fxml"));
+
+            try {
+                Loader.load();
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+
+            IntroPageAdmin introPageAdmin = Loader.getController();
+            introPageAdmin.setAdminId(adminId);
+            introPageAdmin.setUserRole(getUserRole());
```

Comment:
This is returning `null` since we haven't set it from its parent page i.e. IntroPageAdmin. Kindly check the other comment